### PR TITLE
fix: Set pillarTrackers state during onChange

### PR DESCRIPTION
**Pull request by bruce-glazier**
## Changes

  - Fixes consistency issue for the pillar trackers where a changed value (like target) will not persist through screen navigation.
  - Fixes rounding issue when selecting a fractional value via the target selector

### CODE REVIEW

**Review by markdlv — DISMISSED** [review]

File: src/components/TrackTile/TrackerDetails/coerce-to-nonnegative-value.ts [hunk 1]
```diff
@@ -8,5 +8,5 @@ export const coerceToNonnegativeValue = (num: string, defaultNum: number) => {
     return defaultNum;
   }
 
-  return Math.round(targetNum);
```
Comment (bruce-glazier) [inline src/components/TrackTile/TrackerDetails/coerce-to-nonnegative-value.ts:11]: I don't see any reason why we would need to be rounding this. Perhaps we did not allow fractional targets in the past.

Comment (markdlv) [reply src/components/TrackTile/TrackerDetails/coerce-to-nonnegative-value.ts:11]: @sternetj do you have any recollection?  As long as we have display logic to ensure we never show something like 1.3333333, I agree this should be fine.

**Review by markdlv — APPROVED** [review]

File: src/components/TrackTile/TrackerDetails/coerce-to-nonnegative-value.ts (hunk 1 above)
Comment (bruce-glazier) [reply src/components/TrackTile/TrackerDetails/coerce-to-nonnegative-value.ts:11]: I can't think of anywhere which we compute these values in a way that could result in that many digits.

Comment (sternetj) [reply src/components/TrackTile/TrackerDetails/coerce-to-nonnegative-value.ts:11]: I bet this was to handle when we had a number input for selecting your target. A user could enter something like `13e-1` which is a valid JS number but not a whole number.

It seems like it is still used on the tracker value. Was this causing you problems @bruce-glazier

File: src/components/TrackTile/hooks/useTrackers.ts [hunk 2]
```diff
@@ -42,35 +42,71 @@ export const useTrackers = (trackersToUse?: Trackers) => {
         return;
       }
 
+      const pillarTileTrackers = installedMetrics.filter(
+        (t) => t.system === TRACKER_PILLAR_CODE_SYSTEM,
+      );
```
Comment (sternetj) [inline src/components/TrackTile/hooks/useTrackers.ts:47]: Should we pass the context to this hook so we can filter by the context system and not have to have the duplicate logic?

File: src/components/TrackTile/TrackerDetails/coerce-to-nonnegative-value.ts (hunk 1 above)
Comment (bruce-glazier) [reply src/components/TrackTile/TrackerDetails/coerce-to-nonnegative-value.ts:11]: We allow for selecting values such as 0.25, 0.5, 0.75 for units like hours and we use this function to convert from string to number for storing the value. It was rounding those to the nearest whole number.

File: src/components/TrackTile/hooks/useTrackers.ts (hunk 2 above)
Comment (bruce-glazier) [reply src/components/TrackTile/hooks/useTrackers.ts:47]: Sure, I'll take a look at that.

File: src/components/TrackTile/TrackerDetails/coerce-to-nonnegative-value.ts (hunk 1 above)
Comment (sternetj) [reply src/components/TrackTile/TrackerDetails/coerce-to-nonnegative-value.ts:11]: Gotcha, that makes sense. Sounds like its safe to remove

File: src/components/TrackTile/hooks/useTrackers.ts (hunk 2 above)
Comment (bruce-glazier) [reply src/components/TrackTile/hooks/useTrackers.ts:47]: I don't think the juice is worth the squeeze here as it results in a lot more lines of added code passing valuesContext around.

**Pull request merged**